src/encoder/plugins/OpusEncoderPlugin.cxx

```diff
@@ -183,24 +190,25 @@ OpusEncoder::~OpusEncoder()
 void
 OpusEncoder::DoEncode(bool eos)
 {
-	assert(buffer_position == buffer_size);
+	if (!eos)
+		assert(buffer_position == buffer_size);
```

Comment:
You shouldn't call a macro after an `if` statement without curly braces, because this is unsafe; and it doesn't make sense to check the `eos` flag in the non-debug build. It's better to move that check into the `assert()`.

Reply:
will do